From: Hilko Bengen <bengen@debian.org>
Date: Tue, 23 Jul 2019 14:43:21 +0200
Description: Add --with-ebpf-includes parameter
 This patch fixes a FTBFS error on the i386 architecture by setting proper
 include directories. The error is caused by improper directory names e.g.
 '-I/usr/include/i386-linux-gnu' vs. '-I/usr/include/i686-linux-gnu/'.
Last-Update: 2025-07-11
Bug: https://redmine.openinfosecfoundation.org/issues/3097

---
 configure.ac     | 4 ++++
 ebpf/Makefile.am | 3 +--
 2 files changed, 5 insertions(+), 2 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -508,7 +508,11 @@
                     AC_SUBST(LLC)
                   ],
                   [AC_MSG_ERROR([clang needed to build ebpf files])])
-            AC_MSG_CHECKING([libbpf has bpf/bpf_helpers.h])
+            AC_ARG_WITH(ebpf_includes,
+                    [  --with-ebpf-includes=DIR  include directory for building eBPF programs],
+                    [AC_SUBST([ebpf_includes],["$withval"])],
+                    [AC_SUBST([ebpf_includes],["/usr/include/${build_alias}"])])
+	    AC_MSG_CHECKING([libbpf has bpf/bpf_helpers.h])
             AC_COMPILE_IFELSE(
                 [AC_LANG_PROGRAM(
                     [
--- a/ebpf/Makefile.am
+++ b/ebpf/Makefile.am
@@ -4,7 +4,7 @@
 if BUILD_EBPF
 
 # Maintaining a local copy of UAPI linux/bpf.h
-BPF_CFLAGS = -Iinclude
+BPF_CFLAGS = -Iinclude -I$(ebpf_includes)
 
 BPF_TARGETS  = lb.bpf
 BPF_TARGETS += filter.bpf
@@ -19,7 +19,6 @@
 $(BPF_TARGETS): %.bpf: %.c
 #      From C-code to LLVM-IR format suffix .ll (clang -S -emit-llvm)
 	${CLANG} -Wall $(BPF_CFLAGS) -O2 -g \
-		-I/usr/include/$(build_cpu)-$(build_os)/ \
 		-D__KERNEL__ -D__ASM_SYSREG_H \
 		-target bpf -S -emit-llvm $< -o ${@:.bpf=.ll}
 #      From LLVM-IR to BPF-bytecode in ELF-obj file
